app/models/category.rb

```diff
@@ -0,0 +1,26 @@
+class Category < ApplicationRecord
+  self.primary_key = "name"
+
+  def summary
+    @summary ||= summary_query.group_by { |summary| [summary.month, summary.currency] }.map do |key, value|
+      expected_movement = value.detect { |summary| summary.expected_movement }
+      unexpected_movement = value.detect { |summary| !summary.expected_movement }
+
+      {
+        month_currency: "#{key[0]} (#{key[-1]})",
+        expected_movement: (expected_movement.present? ? expected_movement.sum : 0),
+        unexpected_movement: (unexpected_movement.present? ? unexpected_movement.sum : 0),
```

Comment:
I'm wondering if it would be better to bring these totals from the DB view instead of handling them here.
I mean, instead of we have:

month | currency | sum | expected_movement
-------- | ------------ | ----- | ----------------------------
"2021-02" | BRL | 10.0 | true
"2021-02" | NZD | 20.0 | false
"2021-02" | NZD | 50.0 | true

We might have something like:

month | currency | total_expected | total_unexpected
-------- | ------------ | ------------------- | -----------------------
"2021-02" | BRL | 10.0 | 0.0
"2021-02" | NZD | 50.0 | 20.0

This way we ensure only one line per `month + currency`. :thinking:

Reply:
Yes, it would ideal, but my knowledge on SQL is limited :joy: